pkg/ekcoops/poll.go

```diff
@@ -2,30 +2,54 @@ package ekcoops
 
 import (
 	"context"
+	"k8s.io/client-go/dynamic"
 	"time"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // Poll will run the reconcile function on an interval
 func (o *Operator) Poll(ctx context.Context, interval time.Duration) {
+	logger := o.controller.Log
 	ticker := time.NewTicker(interval)
 
+	dynamicClient, err := dynamic.NewForConfig(o.controller.Config.ClientConfig)
+	if err != nil {
+		logger.Error("Failed creating Kubernetes client.", err)
+	}
+
+	prometheusClient := dynamicClient.Resource(prometheusGvr).Namespace("monitoring")
+	alertManagerClient := dynamicClient.Resource(alertManagerGvr).Namespace("monitoring")
+
+	lastNodeCount := 0
 	i := 0
 	for {
 		select {
 		case <-ticker.C:
 			nodeList, err := o.client.CoreV1().Nodes().List(metav1.ListOptions{})
 			if err != nil {
-				o.log.Infof("Skipping reconcile: failed to list nodes: %v", err)
+				logger.Infof("Skipping reconcile: failed to list nodes: %v", err)
 				continue
 			}
 			doFullReconcile := i%60 == 0
 			err = o.Reconcile(nodeList.Items, doFullReconcile)
 			if err != nil {
-				o.log.Infof("Reconcile failed: %v", err)
+				logger.Infof("Reconcile failed: %v", err)
 				continue
 			}
+
+			currentNodeCount := len(nodeList.Items)
+			if currentNodeCount != lastNodeCount {
```

Comment:
this is a javascripty syntax :). how about moving this logic to the `prometheusAutoscaler` function, and have the syntax be a bit more like this instead?
```go
currentNodeCount := len(nodeList.Items)
if currentNodeCount == lastNodeCount {
	return nil
}

prometheus, _ := prometheusClient.Get("k8s", metav1.GetOptions{})
alertManager, _ := alertManagerClient.Get("prometheus-alertmanager", metav1.GetOptions{})
if prometheus == nil || alertManager == nil {
	return nil
}

<actual autoscaler function logic goes here>
```